### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by tanmayb123**

### CODE REVIEW

File: powerai_serverside/datasequence.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import os
```
Comment (stevemar) [inline powerai_serverside/datasequence.py:3]: Add an apache license header for header

File: powerai_serverside/datasequence.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+import tensorflow as tf
```
Comment (stevemar) [inline powerai_serverside/datasequence.py]: imports should be alphabetized and groups by standard lib and 3rd party...

```
import math
import os
import random

from keras.preprocessing.image import img_to_array
from keras.preprocessing.image import load_img
from keras.utils import Sequence
import numpy as np
import tensorflow as tf
```

File: powerai_serverside/datasequence.py (hunk 1 above)
Comment (stevemar) [reply powerai_serverside/datasequence.py:3]: if code was in the public domain then give a link to it.

File: powerai_serverside/datasequence.py (hunk 2 above)
Comment (stevemar) [reply powerai_serverside/datasequence.py]: Also, don't import classes `from keras.utils import Sequence`, use `from keras import utils` and use the module.

File: powerai_serverside/download_extract_data.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+wget http://memorability.csail.mit.edu/lamem.tar.gz
```
Comment (stevemar) [inline powerai_serverside/download_extract_data.sh]: this doesn't really need to be a shell script...

File: powerai_serverside/train_nasnet.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from keras import backend as K
```
Comment (stevemar) [inline powerai_serverside/train_nasnet.py:1]: sys and muli-gpu-model are not used

File: powerai_serverside/train_nasnet.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+train_pd = pd.read_csv("splits/train_1.txt")
+test_pd = pd.read_csv("splits/test_1.txt")
+batch_size = 32
+
+lr_callback = ReduceLROnPlateau(monitor='val_loss', factor=0.2, patience=2, min_lr=0.003)
+tb_callback = TensorBoard()
+mc_callback = ModelCheckpoint(filepath='current_best.hdf5', verbose=1, save_best_only=True)
+
+model.fit_generator(DataSequence(train_pd, "./images", batch_size=batch_size), validation_data=DataSequence(test_pd, "./images", batch_size=batch_size), epochs=20, verbose=1, use_multiprocessing=True, workers=80, steps_per_epoch=len(train_pd) // batch_size, validation_steps=len(test_pd) // batch_size, callbacks=[lr_callback, tb_callback, mc_callback])
```
Comment (stevemar) [inline powerai_serverside/train_nasnet.py]: this should be better formatted.

File: powerai_serverside/train_nasnet.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+model.save("nasnet_lamem_model.h5")
```
Comment (stevemar) [inline powerai_serverside/train_nasnet.py]: consider using a constant up top.

File: powerai_serverside/train_nasnet.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+from keras.applications.nasnet import NASNetMobile
+from keras.callbacks import ReduceLROnPlateau, TensorBoard, ModelCheckpoint
+from datasequence import DataSequence
+import pandas as pd
+import sys
+
+def euc_dist_keras(y_true, y_pred):
+    return K.sqrt(K.sum(K.square(y_true - y_pred), axis=-1, keepdims=True))
+
+model = NASNetMobile()
```
Comment (stevemar) [inline powerai_serverside/train_nasnet.py]: comments here would go a long way in helping maintainers understand whats going on

File: webapp/public/README.md [hunk 8]
```diff
@@ -0,0 +1 @@
+This directory will contain all of the images uploaded to the webapp.
```
Comment (stevemar) [inline webapp/public/README.md:1]: where the images?

**Review by stevemar — CHANGES_REQUESTED** [review]

File: webapp/public/README.md (hunk 8 above)
Comment (tanmayb123) [reply webapp/public/README.md:1]: The images are uploaded by the user. Empty by default.
(every time you use the web app, your image will be uploaded here)

**Review by stevemar — APPROVED** [review]

**Pull request merged**